services/pushover.py

```diff
@@ -77,9 +93,33 @@ def plugin(srv, item):
     if callback is not None:
         params['callback'] = callback
 
+    filepayload=None
+    try:
+        srv.logging.debug("Loading mesasgejson")
+
+        messagejson = json.loads(message)
+        message = messagejson["message"]
+        if 'imageurl' in messagejson:
+            srv.logging.debug("Image url is in json object %s" % messagejson["imageurl"])
+            filepayload = { "attachment": ("image.jpg",requests.get(messagejson["imageurl"], stream=True).raw, "image/jpeg") }
+        elif 'image' in messagejson:
+            srv.logging.debug("Image is in json object %s" % messagejson["image"])
+            filepayload = { "attachment": ("image.jpg",base64.decodestring(messagejson["image"]), "image/jpeg") }
+        else:
+            srv.logging.debug("No image could be found,")
+			
+
+    except JSONDecodeError, jsone:
```

Comment:
I think this should be a bit more forgiving of unexpected or invalid payloads. E.g. if you don't get a JSON payload then just pass the message as-is, this is how it currently works. If you get a JSON payload then check for the `message` attribute - if none found then pass the whole payload as-is. If there is a `message` attribute then check for your two image options, and if neither, or one of them fail to decode or resolve, then log a warning but still pass the message.

So we try and get a message delivered at all costs. Any issues get logged as warnings, but proceed. Sound reasonable?

Reply:
Definitely